test/Microsoft.AspNetCore.Mvc.ViewFeatures.Test/Rendering/HtmlHelperDropDownListExtensionsTest.cs

```diff
@@ -0,0 +1,232 @@
+                "</select>";
+            var metadataProvider = new EmptyModelMetadataProvider();
+            var helper = DefaultTemplatesUtilities.GetHtmlHelper(new ViewDataDictionary<TestModel>(metadataProvider));
+            helper.ViewContext.ClientValidationEnabled = false;
+            helper.ViewData.Model = new TestModel { Property1 = BasicSelectList };
+
+            // Act
+            var dropDownListResult = helper.DropDownList("Property1");
+
+            // Assert
+            Assert.Equal(expectedHtml, HtmlContentUtilities.HtmlContentToString(dropDownListResult));
+        }
+
+        [Fact]
+        public void DropDownList_UsesSpecifiedExpressionAndOptionLabel()
+        {
+            // Arrange
+            var expectedHtml = "<select id=\"HtmlEncode[[Property1]]\" name=\"HtmlEncode[[Property1]]\">" +
+                "<option value=\"\">HtmlEncode[[--select--]]</option>" + Environment.NewLine +
+                "<option value=\"HtmlEncode[[0]]\">HtmlEncode[[Zero]]</option>" + Environment.NewLine +
+                "<option value=\"HtmlEncode[[1]]\">HtmlEncode[[One]]</option>" + Environment.NewLine +
+                "<option value=\"HtmlEncode[[2]]\">HtmlEncode[[Two]]</option>" + Environment.NewLine +
+                "<option value=\"HtmlEncode[[3]]\">HtmlEncode[[Three]]</option>" + Environment.NewLine +
+                "</select>";
+            var metadataProvider = new EmptyModelMetadataProvider();
+            var helper = DefaultTemplatesUtilities.GetHtmlHelper(new ViewDataDictionary<TestModel>(metadataProvider));
+            helper.ViewContext.ClientValidationEnabled = false;
+            helper.ViewData.Model = new TestModel { Property1 = BasicSelectList };
+
+            // Act
+            var dropDownListResult = helper.DropDownList("Property1", "--select--");
+
+            // Assert
+            Assert.Equal(expectedHtml, HtmlContentUtilities.HtmlContentToString(dropDownListResult));
+        }
+
+        [Fact]
+        public void DropDownList_UsesSpecifiedExpressionAndSelectList()
+        {
+            // Arrange
+            var expectedHtml = "<select id=\"HtmlEncode[[Property1]]\" name=\"HtmlEncode[[Property1]]\">" +
+                "<option value=\"HtmlEncode[[4]]\">HtmlEncode[[Four]]</option>" + Environment.NewLine +
+                "<option value=\"HtmlEncode[[5]]\">HtmlEncode[[Five]]</option>" + Environment.NewLine +
+                "</select>";
+            var metadataProvider = new EmptyModelMetadataProvider();
+            var helper = DefaultTemplatesUtilities.GetHtmlHelper(new ViewDataDictionary<TestModel>(metadataProvider));
+            helper.ViewContext.ClientValidationEnabled = false;
+            helper.ViewData.Model = new TestModel { Property1 = BasicSelectList };
+
+            // Act
+            var dropDownListResult = helper.DropDownList("Property1", new List<SelectListItem>
```

Comment:
Never do this. Give `Property1` a real value when passing the select list in. Muck with the value and you'll see `selected="selected"` start to show up in the HTML.